iothub/device/src/Authentication/HsmAuthentication/EdgeModuleAuthenticationWithHsm.cs

```diff
@@ -10,13 +10,13 @@ namespace Microsoft.Azure.Devices.Client.HsmAuthentication
     /// <summary>
     /// Authentication method that uses HSM to get a SAS token.
     /// </summary>
-    internal class ModuleAuthenticationWithHsm : ClientAuthenticationWithTokenRefresh
+    internal class EdgeModuleAuthenticationWithHsm : ClientAuthenticationWithTokenRefresh
```

Comment:
this is a special case applicable for the edge module only, so rename this class